### ISSUE TITLE
Make repo Maven compatible

### ISSUE THREAD

**Issue by ssardina** [SWI-Prolog/packages-jpl#48]
In most cases, the Java side of JPL may be updated/improved without the C side being touched.  This means one may want a more recent `jpl.jar` than the  official one in repos or SWI stable distribution.

Converting this repo into a Maven-compatible one, would ease this a lot. And we will be able to offer JPL Java API as an artifact on the web (without updating the whole SWI system). 

I would also make it better in terms of SE, as Maven is very standard today in the Java community. Most Java IDE (IntelliJ or Ecllipse) support Maven projects and make everything much easier.

This should be fairly easy, but part of the job is to refactor the sources into the standard Maven structure (there were some issues when doing that with CMAKE)

**Comment by ssardina** [comment]
Work has started in branch maven-new (29674db0abf9214b22b2a7a892e6cff451186131)

Source has been moved to `src/main/java` and POM file works.

However, CMAKE fails to find the jar file for testing:

```
[ssardina@Thinkpad-X1 build]$ ctest -V -R jpl:java_in_prolog

63: ERROR: /home/ssardina/git/soft/prolog/swipl-devel-maven.git/packages/jpl/jpl.pl:4338:
63: ERROR:    /home/ssardina/git/soft/prolog/swipl-devel-maven.git/packages/jpl/jpl.pl:4338: Initialization goal raised exception:
63: ERROR:    source_sink `jar('jpl.jar')' does not exist
63: ERROR: /home/ssardina/git/soft/prolog/swipl-devel-maven.git/packages/jpl/test_jpl.pl:55:
63: ERROR:    Exported procedure jpl:jpl_c_lib_version/1 is not defined
63: % PL-Unit: jpl 
```

**Comment by JanWielemaker** [comment]
Guess you need to update this (snippet from `git grep  -n file_search_path`)

```
test_jpl.pl:45::- asserta(user:file_search_path(jar, '.')).
test_jpl.pl:46::- asserta(user:file_search_path(jar, foreign('src/java'))).
test_jpl.pl:47::- asserta(user:file_search_path(jar, foreign('.'))).
```

**Comment by ssardina** [comment]
Thanks Jan. Yes, yes, I have solved that and many other paths and files. It is almost cooked and so much nicer structure. 

Once the other PR #61  is merged into master I will go for this one, it is almost finished.

This will allow:

1. Development as a Maven project, which is much nicer for Java and much more "standard" these days, with all the IDEs providing good support for that.
2. More importantly, allow accessing updated `jpl.jar` without having to compile the whole system, as applications can gather the updated `jpl.jar` via Maven dependency. Almost 99% of changes in JPL do not require changes in the C code.

**Comment by JanWielemaker** [comment]
Good.  I do assume this doesn't break the CMake build, no?   Preferably also doesn't change any _installation_ locations as that always means trouble with packaging, people not finding stuff, etc.

**Comment by ssardina** [comment]
Right. I am heavily testing CMAKE as it CMAKE with Java can be tricky. I had to do some changes to the CMAKE files because application sources and testing sources are kept separately under Maven structure (`src/main/java` and `src/test/java`) but it is all working well.

There will be no changes to the installation, same things in the same dirs, correct. 

I am covering all this, but thanks for reminding me of all this, great.
